Request

yjch91 opened the pull request:
#8 리퀘스트에 로우 데이터를  헤더를 파싱하고 바디 캐이스에 따라 바디도 파싱함 더이상 읽을 필요없을 때 false를 반환, true를 반환하면 더 읽을게 남은 상태

Commit:
[FEAT]: Request Default Class Declare

Request Default 클래스 작성
기본 Header 멤버 변수, Getter/Setter, Default Header 생성 Method

Commit:
[FEAT]: Request Raw to Object

Raw Request Header 를 Object화 파싱 초안 작성
예외 사항, 문법 특수 사항, 데이터 구조 재 고안 필요

Commit:
[REFACTOR]: Response Main
Main Test

Commit:
[REFACTOR]: Method Rule Change
Need to about TCP, Chunked Rule Check

Commit:
[FEAT] Request header body parsing (not finished)

yjch91 committed:
[FEAT] Request 뒤에 부분 raw_request 에 남게 수정

pearan2 commented:
generateRequest 함수와 setter / getter 를 제외하면 private 으로 가도 될 것 같습니다.

Likilee commented on includes/Request.hpp:
```diff
@@ -0,0 +1,115 @@
+		void	initRequest(void);
+
+		int		generateRequest(void);
+		void	generateStartLine(void);
+		void	generateRequestHeader(void);
+		void	generateRequestBody(void);
+
+		void	parseMethod(void);
+		void	parseUri(void);
+		void	parseHttpVersion(void);
+
+		void	parseAcceptCharsets(void);
+		void	parseAcceptLanguage(void);
+		void	parseAuthorization(void);
+		void	parseContentLength(void);
+		void	parseContentType(void);
+		void	parseDate(void);
+		void	parseHost(void);
+		void	parseReferer(void);
+		void	parseTransferEncoding(void);
+		void	parseUserAgent(void);
```
외부에서 사용하는 generateRequest 하위 메소드들을 private으로 이동시키면 좋을거 같습니당 :)

yjch91 committed:
[FIX] 클리어 빼먹어서 추가함 temp_body.clear

Likilee commented on includes/Request.hpp:
```diff
@@ -0,0 +1,115 @@
+
+		void	initRequest(void);
+
+		int		generateRequest(void);
```
generate 가 정확한 표현일 수 있지만, 영어고자인 저에게는 구글 번역을 써야할 수준의 단어라서, make나 create정도의 중학영어 수준의 naming은 어떨까요? ㅎㅎ

Likilee reviewed (commented):
LGTM!

pearan2 commented:
LGTM ~

Commit:
[FIX] : Request class

- Request 멤버 메소드(generate*) 이름 변경 (make*)
- (make*) 메소드들 private로 변경
- tryMakeRequest 메서드 및 isComplete 메서드 반환값 변경
- 오타 및 작은 이슈 수정

yjch91 reviewed (commented):
LGTM!!!!!!!!!!!!!!!!!!

Commit:
[FIX] fix Request class tryMake* method name to make*

pearan2 commented:
LGTM !!!!!!!!
